connector_ecommerce/sale.py

```diff
@@ -0,0 +1,321 @@
+                    # action, sales orders use the action_cancel method.
+                    if state in wkf_states:
+                        order.signal_workflow('cancel')
+                    elif state in action_states:
+                        order.action_cancel()
+                    else:
+                        raise ValueError('%s should not fall here.' % state)
+                except (osv.except_osv, osv.orm.except_orm,
+                        exceptions.Warning):
+                    # the 'cancellation_resolved' flag will stay to False
+                    message = _("The sales order could not be automatically "
+                                "canceled.") + resolution_msg
+                else:
+                    message = _("The sales order has been automatically "
+                                "canceled.")
+            else:
+                # shipping_except, invoice_except, ...
+                # can not be canceled from the view, so assume that it
+                # should not be canceled here neiter, exception to
+                # resolve
+                message = _("The sales order could not be automatically "
+                            "canceled for this status.") + resolution_msg
+            order.message_post(body=message)
+
+    @api.multi
+    def _log_canceled_in_backend(self):
+        message = _("The sales order has been canceled on the backend.")
+        self.message_post(body=message)
+        for order in self:
+            message = _("Warning: the origin sales order %s has been canceled "
+                        "on the backend.") % order.name
+            if order.picking_ids:
+                order.picking_ids.message_post(body=message)
+            if order.invoice_ids:
+                order.invoice_ids.message_post(body=message)
+
+    @api.model
+    def create(self, values):
+        order = super(SaleOrder, self).create(values)
+        if values.get('canceled_in_backend'):
+            order._log_canceled_in_backend()
+            order._try_auto_cancel()
+        return order
+
+    @api.multi
+    def write(self, values):
+        result = super(SaleOrder, self).write(values)
+        if values.get('canceled_in_backend'):
+            self._log_canceled_in_backend()
+            self._try_auto_cancel()
+        return result
+
+    @api.multi
+    def action_cancel(self):
+        res = super(SaleOrder, self).action_cancel()
+        for sale in self:
+            # the sale order is canceled => considered as resolved
+            if (sale.canceled_in_backend and
+                    not sale.cancellation_resolved):
+                sale.write({'cancellation_resolved': True})
+        return res
+
+    @api.multi
+    def ignore_cancellation(self, reason):
+        """ Manually set the cancellation from the backend as resolved.
+
+        The user can choose to keep the sale order active for some reason,
+        so it just push a button to keep it alive.
```

Comment:
"[...] it is just needed to push a button [...]"
or
"[...] it only requires to push a button [...]"